tfhe/src/integer/server_key/radix/add.rs

```diff
@@ -119,17 +119,34 @@ impl ServerKey {
                 >= (left_block.message_modulus.0 * left_block.carry_modulus.0)
             {
                 // We would exceed the block 'capacity'
-                return false;
+                return Err(CheckError::CarryFull {
+                    degree: Degree(degree_after_add + preceding_block_carry),
+                    max_degree: MaxDegree(
+                        left_block.message_modulus.0 * left_block.carry_modulus.0 - 1,
+                    ),
+                });
             }
+
+            let extracted_carry_noise_level = if first_block {
+                NoiseLevel::ZERO
+            } else {
+                NoiseLevel::NOMINAL
+            };
```

Comment:
Or init `extracted_carry_noise_level`  to zero before the loop and at the end of the loop set it to  Nominal

Reply:
Done